flink-cdc-common/src/main/java/org/apache/flink/cdc/common/utils/SchemaMergingUtils.java

```diff
@@ -169,20 +173,52 @@ public static Schema getLeastCommonSchema(
     }
 
     /** Merge compatible schemas. */
-    public static Schema getCommonSchema(List<Schema> schemas) {
-        if (schemas.isEmpty()) {
-            return null;
-        } else if (schemas.size() == 1) {
-            return schemas.get(0);
+    public static Schema getCommonSchema(LinkedHashSet<Schema> schemas) {
```

Comment:
This is a breaking API change since the parameter type constraints are getting stricter. I think changing List => Collection is acceptable since it's getting wider.